### PR TITLE
Remaining requirements

### PR DESCRIPTION
**Pull request by Taracuda**
This PR covers the remaining coding requirements as follows:

- 1.5
- 1.6
- 1.8
- 1.10
- 1.12
- 1.22

Once merged, our CollectR website application will be at full functioning capacity! UI cleanup will come later, prior to demo.

### CODE REVIEW

File: src/components/CollectionBox.tsx [hunk 1]
```diff
@@ -44,9 +43,8 @@ export const CollectionBox: React.FC<CollectionBoxProps> = ({collectionName, col
           }
           {!isEditing && (
               <>
-            <Link to={`/collection/${collectionId}`}>
```
Comment (piperlincoln) [inline src/components/CollectionBox.tsx:47]: Why was this removed? How will the user get inside the collection?

File: src/components/FakeCollectionStore.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    public getCollections = (): CollectionModel[] => {
+        return this.collections;
+    }
+
+    public setCollections = (newCollections: CollectionModel[]): void => {
+        this.collections = newCollections;
+    }
+
+    public setFavorite = (favoritedItem: CollectionItemModel) => {
```
Comment (piperlincoln) [inline src/components/FakeCollectionStore.tsx:16]: Can a favorite get removed?

**Review by piperlincoln — CHANGES_REQUESTED** [review]
Looks great Tara! Looks like the pipeline had a couple errors, but they should be quick to fix!

File: src/components/CollectionBox.tsx (hunk 1 above)
Comment (Taracuda) [reply src/components/CollectionBox.tsx:47]: That piece is actually happening now in PageLayoutComponent.tsx at about line 121

File: src/components/FakeCollectionStore.tsx (hunk 2 above)
Comment (Taracuda) [reply src/components/FakeCollectionStore.tsx:16]: Not with this current application, no. It will still cover our requirements though, given that there were none mentioning the removal of a favorite. Obviously, without a time crunch I'd feel better about figuring that out, but as we stand on requirements, we're in good shape.

File: src/Database/CollectionStore.ts [hunk 3]
```diff
@@ -35,6 +35,18 @@ class CollectionStore {
...
+        const favorites: CollectionItemModel[] = JSON.parse(localStorage.getItem("favorites") || "[]");
+
+        favorites.push(favoritedItem);
+        localStorage.setItem("favorites", JSON.stringify(favorites));
+    }
+
+    public getFavorites = (): CollectionItemModel[] => {
+        const favorites: CollectionItemModel[] = JSON.parse(localStorage.getItem("favorites") || "[]");
+        return favorites;
+    }
```
Comment (firicano33) [inline src/Database/CollectionStore.ts:49]: Hi Tara.  I see here that favorites functionality has been added.   Looks good.

**Review by firicano33 — APPROVED** [review]

**Review by piperlincoln — APPROVED** [review]
Thanks for answering my questions! Looks great!

**Comment by hsantos940** [comment]
I'll get to work on testing these.
